Microsoft.Identity.Web/Client/TokenAcquisition.cs

```diff
@@ -186,6 +177,15 @@ public async Task<string> GetAccessTokenOnBehalfOfUser(HttpContext context, IEnu
 
                 return result.AccessToken;
             }
+            else
+            {
+                var result = await application.AcquireTokenSilent(
+                    scopes.Except(scopesRequestedByMsalNet),
+                    accounts.FirstOrDefault())
```

Comment:
is that correct? are we guaranteed that first account matches current user?

Reply:
@yordan-msft - token cache is persisted per access token, in this case, there will always only be one account in the cache.

Reply:
are we guaranteed? Would it be safer to lookup the account based on the access token, just in case?

Reply:
@yordan-msft. for the user token cache the recommendation is to have one cache per account (and now one cache per access token used to call the Web API)